src/utils/frameworks-api.ts

```diff
@@ -1,24 +1,33 @@
-import { mkdir } from 'fs/promises'
+import { access, mkdir, readFile } from 'node:fs/promises'
 import { resolve } from 'node:path'
 
+import { mergeConfigs } from '@netlify/config'
+
+import type { NetlifyOptions } from '../commands/types.js'
+
 interface FrameworksAPIPath {
   path: string
   ensureExists: () => Promise<void>
+  exists: () => Promise<boolean>
 }
 
+export type FrameworksAPIPaths = ReturnType<typeof getFrameworksAPIPaths>
+
 /**
  * Returns an object containing the paths for all the operations of the
- * Frameworks API. Each key maps to an object containing a `path` property
- * with the path of the operation and a `ensureExists` methos that creates
- * the directory in case it doesn't exist.
+ * Frameworks API. Each key maps to an object containing a `path` property with
+ * the path of the operation, an `exists` method that returns whether the path
+ * exists, and an `ensureExists` methos that creates it in case it doesn't.
  */
```

Comment:
😂 this fixes a methos → method typo but introduces a different methos typo!
```suggestion
 * Frameworks API. Each key maps to an object containing a `path` property with
 * the path of the operation, an `exists` method that returns whether the path
 * exists, and an `ensureExists` method that creates it in case it doesn't.
 */
```

Reply:
Good grief!